pymobiledevice3/tunneld.py

```diff
@@ -437,13 +437,39 @@ def _run_app(self) -> None:
 
 async def async_get_tunneld_devices(tunneld_address: Tuple[str, int] = TUNNELD_DEFAULT_ADDRESS) \
         -> List[RemoteServiceDiscoveryService]:
+    tunnels = _list_tunnels(tunneld_address)
+    return await _create_rsds_from_tunnels(tunnels)
+
+
+def get_tunneld_devices(tunneld_address: Tuple[str, int] = TUNNELD_DEFAULT_ADDRESS) \
+        -> List[RemoteServiceDiscoveryService]:
+    return get_asyncio_loop().run_until_complete(async_get_tunneld_devices(tunneld_address))
+
+
+async def async_get_tunneld_device_by_udid(udid: str, tunneld_address: Tuple[str, int] = TUNNELD_DEFAULT_ADDRESS) \
+        -> List[RemoteServiceDiscoveryService]:
+    tunnels = _list_tunnels(tunneld_address)
+    if udid not in tunnels:
+        return []
+    return await _create_rsds_from_tunnels({udid: tunnels[udid]})
+
+
+def get_tunneld_device_by_udid(udid: str, tunneld_address: Tuple[str, int] = TUNNELD_DEFAULT_ADDRESS) \
+        -> List[RemoteServiceDiscoveryService]:
```

Comment:
Returning a list here is more than what this function is expected to. Let's return here a single `RemoteServiceDiscoveryService`

Reply:
Will do, planning to return None if there is no match. This will support scenario of only one <iOS17 iPhone connected to setup.

I need to consider providing >=iOS17 support while maintaining compatibility to test in <iOS17.  The libraries I work on have been written to work for any number of iPhones connected to setup, which is not known ahead of time.

Below is an example of the use case.

```python
# `pymobiledevice3 remote tunneld` already running in a separate process

def get_service_provider(udid):
  service_provider = get_tunneld_device_by_udid(udid)
  if service_provider is None:
    # raise error if not <iOS17
    service_provider = create_using_usbmux(serial=udid)
  return service_provider
```